plugins/BEdita/Core/src/Model/Table/ObjectPermissionsTable.php

```diff
@@ -86,4 +106,69 @@ public function buildRules(RulesChecker $rules): RulesChecker
             ->add($rules->existsIn(['role_id'], 'Roles'))
             ->add($rules->existsIn(['created_by'], 'CreatedByUsers'));
     }
+
+    /**
+     * Check if object permission is editable before saving.
+     *
+     * @param \Cake\Event\EventInterface $event Fired event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity data.
+     * @return void
+     */
+    public function beforeSave(EventInterface $event, EntityInterface $entity): void
+    {
+        if (!$this->isEditable($entity)) {
+            throw new ForbiddenException(__d('bedita', 'Save object permission is forbidden for user'));
+        }
+    }
+
+    /**
+     * Check if object permission is editable before deleting.
+     *
+     * @param \Cake\Event\EventInterface $event Fired event.
+     * @param \Cake\Datasource\EntityInterface $entity Entity data.
+     * @return void
+     */
+    public function beforeDelete(EventInterface $event, EntityInterface $entity): void
+    {
+        if (!$this->isEditable($entity)) {
+            throw new ForbiddenException(__d('bedita', 'Delete object permission is forbidden for user'));
+        }
+    }
+
+    /**
+     * Check if logged user can create, edit or delete the object permission.
+     *
+     * Object permission is editable if one of below condition is true:
+     * - user belongs to admin role
+     * - no permission is set on the related object
+     * - permissions are set on the related object and user belongs to role that with permission
```

Comment:
Could we use also use `role` priority for that? 
F.i. if a permission is set, it is editable if:

* user has `admin` role
* user has the same role of the permission
* user has a lower priority role than the permission role

Reply:
Perhaps my comment was not clear enough. 
I was not referring to the permissions rules itself (can i modify this object?) but to the permission editing (can I edit this permission?). 
We could use role `priority` to see if a certain role has enough power to edit permission for another role.
But we can implement this later, if necessary.